crates/cli/src/opts/build/zksync.rs

```diff
@@ -1,22 +1,56 @@
 use std::path::PathBuf;
 
-use clap::Parser;
+use clap::{builder::PossibleValue, Parser, ValueEnum};
 use foundry_config::ZkSyncConfig;
-use serde::Serialize;
 
-#[derive(Clone, Debug, Default, Serialize, Parser)]
+/// Represents the zkVM mode setting
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Default)]
+enum Mode {
```

Comment:
There seem to be two things here. One if you run zkEVM or not, one if you compile zksync contract or not. Might be that we want separate binary options for both if there's a use case for all scenarios? Or if one (say don't compile but run makes no sense)  is not useful, have an enum that is `Disabled | CompileOnly | Run` or something like that, where if we should compile or not is derived from that and is not a sepparate setting. In any case this would make sense only for scripts/test and not build so a global `Mode` maybe is not the best past forward. Whatever I am missing the general principles to underline for settings and data structures to override each other as is the case here when `Run | CompileOnly` will force some setting and `Enable` will not so there's a high cognitive overload to understand paths if things override like this, and also make things as local as possible so you do not end up with varables used globally but only apply to specific scenarios.